### PR TITLE
Fixes #1073 Cannot open a v11.2 project in v12

### LINKED ISSUE #1073: Cannot open a v11.2 project in v12
https://github.com/Open-Systems-Pharmacology/OSMOSES/blob/examples/UseCases/V11.2/omeprazole%2040mg%20po%20asian%20EM%20fluvoxamine%2050mg%20po_V11.mbp3

- A v11.2 snapshot has been loaded in PK-Sim v11.2
- A simulation has been saves a pkml
- The pkml simulation has been opened with MoBi v11.2 and the project has been saved
- Loading the project with MoBi V12:

Application:
MoBi® 12.0.43

Sequence contains no matching element

Stack trace:
```
at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source, Func`2 predicate)
   at MoBi.Core.Services.TemplateResolverTask.TemplateBuildingBlockFor[TBuildingBlock](TBuildingBlock buildingBlock)
   at MoBi.Presentation.Tasks.Interaction.InteractionTasksForSimulation.TemplateBuildingBlockFor(IBuildingBlock buildingBlock)
   at MoBi.Presentation.Presenter.Main.SimulationExplorerPresenter.<findChangedTemplateBuildingBlocks>b__22_0(IBuildingBlock x)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at MoBi.Presentation.Presenter.Main.SimulationExplorerPresenter.refreshDisplayedSimulation(IMoBiSimulation simulation)
   at MoBi.Presentation.Presenter.Main.SimulationExplorerPresenter.addClassifiableSimulationToTree(ITreeNode`1 classificationNode, ClassifiableSimulation classifiableSimulation)
   at OSPSuite.Presentation.Presenters.AbstractExplorerPresenter`2.AddClassifiableToTree[T](T classifiable, RootNodeType rootNodeType, Func`3 addClassifiableToTree)
   at MoBi.Presentation.Presenter.Main.SimulationExplorerPresenter.addClassifiableSimulationToSimulationRootFolder(ClassifiableSimulation classifiableSimulation)
   at MoBi.Presentation.Presenter.Main.SimulationExplorerPresenter.<AddProjectToTree>b__3_0(ClassifiableSimulation x)
   at OSPSuite.Utility.Extensions.EnumerableExtensions.Each[T](IEnumerable`1 list, Action`1 action)
   at MoBi.Presentation.Presenter.Main.SimulationExplorerPresenter.AddProjectToTree(MoBiProject project)
   at MoBi.Presentation.Presenter.Main.ExplorerPresenter`2.AddProjectToTree(IProject project)
   at OSPSuite.Presentation.Presenters.AbstractExplorerPresenter`2.Handle(ProjectCreatedEvent eventToHandle)
   at OSPSuite.Utility.Events.EventPublisher.<>c__DisplayClass4_1`1.<PublishEvent>b__1()
   at OSPSuite.Utility.Exceptions.ExceptionManagerBase.Execute(Action action)
```

### PR DESCRIPTION
**Pull request by rwmcintosh**

### CODE REVIEW

**Comment by rwmcintosh** [comment]
When you open a single simulation project with the right number of building blocks your module explorer will be organized as below - with one building block per module.
image

File: src/MoBi.Core/Domain/Extensions/ModuleExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using OSPSuite.Core.Domain;
+
+namespace MoBi.Core.Domain.Extensions
+{
+   public static class ModuleExtensions
+   {
+      public static bool IsTemplateMatchFor(this Module module1, Module module2)
```
Comment (msevestre) [inline src/MoBi.Core/Domain/Extensions/ModuleExtensions.cs:7]: Very unclear what this does. Maybe this should not be an extension but more of a local method. Here we are comparing the name between two modules? Where is this used?

File: src/MoBi.Core/Services/ModuleConfigurationBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+﻿using System.Linq;
+using MoBi.Core.Domain.Model;
+using OSPSuite.Core.Domain;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Core.Services
+{
+   public interface IModuleConfigurationBuilder
```
Comment (msevestre) [inline src/MoBi.Core/Services/ModuleConfigurationBuilder.cs:8]: Shouldn't we move this into some kind of converter namespace ?This is not something that should be used outside of conversion right?

File: src/MoBi.Core/Services/ModuleConfigurationBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      public ModuleConfigurationBuilder(IModuleFactory moduleFactory)
+      {
+         _moduleFactory = moduleFactory;
+      }
+
+      public void BuildSimulationModuleConfigurations(MoBiProject project) => project.Simulations.Each(rebuildModuleConfigurationsFrom);
+
+      private void rebuildModuleConfigurationsFrom(IMoBiSimulation simulation)
+      {
+         // ToList() is needed here because module configurations will be removed by replaceConfiguration
```
Comment (msevestre) [inline src/MoBi.Core/Services/ModuleConfigurationBuilder.cs:26]: GOOD!!

File: src/MoBi.Presentation/Nodes/ModuleConfigurationNode.cs [hunk 4]
```diff
@@ -9,7 +10,7 @@ public class ModuleConfigurationNode : AbstractNode<ModuleConfigurationDTO>
       public ModuleConfigurationNode(ModuleConfigurationDTO moduleConfiguration)
          : base(moduleConfiguration)
       {
-         Id = moduleConfiguration.Module.Id;
+         Id = Guid.NewGuid().ToString();
```
Comment (msevestre) [inline src/MoBi.Presentation/Nodes/ModuleConfigurationNode.cs:13]: explain why are you not using the Id of the module here. Isn't this supposed to be unique? Very suspicious

**Review by msevestre — COMMENTED** [review]
Looks good to me. A few minor things

File: src/MoBi.Core/Services/ModuleConfigurationBuilder.cs (hunk 2 above)
Comment (rwmcintosh) [reply src/MoBi.Core/Services/ModuleConfigurationBuilder.cs:8]: I'll move the existing one too ```ProjectClassificationConverter```

File: src/MoBi.Presentation/Nodes/ModuleConfigurationNode.cs (hunk 4 above)
Comment (rwmcintosh) [reply src/MoBi.Presentation/Nodes/ModuleConfigurationNode.cs:13]: The module Id is unique, but the module can be used any number of times within the simulation explorer. I'll add a comment.

Comment (rwmcintosh) [reply src/MoBi.Presentation/Nodes/ModuleConfigurationNode.cs]: Well, I'm not sure what led me to change that originally, but after thinking about it more and testing, this isn't quite right. The modules are clones of the project module, so they should have unique Id's in the simulation explorer.

**Review by msevestre — APPROVED** [review]

File: src/MoBi.Presentation/Nodes/ModuleConfigurationNode.cs (hunk 4 above)
Comment (msevestre) [reply src/MoBi.Presentation/Nodes/ModuleConfigurationNode.cs]: Exactly. This is not accurate otherwise

**Pull request merged**